regression/run/references/main.f

```diff
@@ -0,0 +1,46 @@
+! Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+! other Shroud Project Developers.
+! See the top-level COPYRIGHT file for details.
+!
+! SPDX-License-Identifier: (BSD-3-Clause)
+! #######################################################################
+!
+! Test code generated by the references test.
+!
+program tester
+  use fruit
+  use iso_c_binding
+  use references_mod
+  implicit none
+  logical ok
+
+  call init_fruit
+
+  call test_arraywrapper
+
+  call fruit_summary
+  call fruit_finalize
+
+  call is_all_successful(ok)
+  if (.not. ok) then
+     call exit(1)
+  endif
+
+contains
+
+  subroutine test_arraywrapper
+    type(ArrayWrapper) arr1
+    real(C_DOUBLE), pointer :: arr(:)
+
+    arr1 = ArrayWrapper_ctor()
+    call arr1%set_size(10)
+    call assert_equals(10, arr1%get_size())
+
+    call arr1%allocate()
+    arr => arr1%get_array()
```

Comment:
I'm sure this is a basic Fortran question, but, for this pointer assignment:
```fortran
arr => arr1%get_array() 
```
I'm assuming the following would be a compiler error:
```fortran
arr = arr1%get_array() 
```
Is that correct, or would the second option work, but have different semantics?

Reply:
The fat assign `=>` assigns to the pointer.  The normal assign, `=`, assigns to the pointee. Think `int *ptr`, it's `ptr = getArray()` vs `*ptr = getArray()`.
The compiler will not complain if `get_array` returns a conformable array (both 1-d).
I made that mistake at first, and got a segfault running since it had a bad pointer value.

Reply:
Good to know, thanks.